webapp/models.py

```diff
@@ -90,43 +90,63 @@ def _process_html(html):
 
 def _replace_notifications(soup):
     """
-    Given some BeutifulSoup of a document,
+    Given some BeautifulSoup of a document,
     replace blockquotes with the appropriate notification markup
     """
 
     with open("templates/_notification.html") as notification_file:
         notification_template = Template(notification_file.read())
 
-    for note in soup.findAll(text=re.compile("ⓘ ")):
-        paragraph = note.parent
-        blockquote = paragraph.parent
-
-        if paragraph.name == "p" and blockquote.name == "blockquote":
-            text_contents = paragraph.encode_contents().decode("utf-8")
-            new_contents = re.sub("^ⓘ ", "", text_contents)
+    for note_string in soup.findAll(text=re.compile("ⓘ ")):
+        first_paragraph = note_string.parent
+        blockquote = first_paragraph.parent
+        last_paragraph = blockquote.findChildren(recursive=False)[-1]
+
+        if first_paragraph.name == "p" and blockquote.name == "blockquote":
+            # Remove extra padding/margin
+            first_paragraph.attrs["class"] = "u-no-padding--top"
```

Comment:
Does this class attr not need a check like the `last_paragraph` below?

Reply:
nope. It will just create it if it doesn't exist, and in fact we know it won't exist in this context. You only need it in the latter case so it doesn't clobber the class we just created if the last element happens to be the same as the first element.